### PR TITLE
Spy arithmetic

### PR DESCRIPTION
**Pull request by pantaray**
Author Guidelines
-----------------
- [X] Is the change set **< 400 lines**?
- [X] Was the code checked for memory leaks/performance bottlenecks?
- [X] Is the code running locally **and** on the ESI cluster?
- [X] Is the code running on all supported platforms?

**Note** tests have not been implemented yet. This is intended as Code Review PR for the used methodology to enable arithmetic operations with Syncopy objects. However, basal functionality tests have been carried out sequentially, concurrently on a local machine and via SLURM. 

Reviewer Checklist
------------------
- [x] Are testing routines present?
- [x] Do **parallel** loops have a set length and correct termination conditions?
- [x] Do objects in the global package namespace perform proper parsing of their input? 
- [x] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [x] Code layout
  - [x] Is the code PEP8 compliant?
  - [x] Does the code adhere to agreed-upon naming conventions?
  - [x] Are keywords clearly named and easy to understand?
  - [x] No commented-out code?
- [ ] Are all docstrings complete and accurate?

### CODE REVIEW

File: syncopy/datatype/base_data.py [hunk 1]
```diff
@@ -738,9 +738,22 @@ def __del__(self):
                 shutil.rmtree(os.path.splitext(self.filename)[0],
                               ignore_errors=True)
 
+    # Support for basic arithmetic operations (no in-place computations supported yet)
     def __add__(self, other):
         return _process_operator(self, other, "+")
 
+    def __sub__(self, other):
+        return _process_operator(self, other, "-")
```
Comment (tensionhead) [inline syncopy/datatype/base_data.py:749]: Overloading with non-member functions probably isn't standard, yet here absolutely the right way.. also looks super clean :+1:

File: syncopy/datatype/methods/arithmetic.py [hunk 2]
```diff
@@ -165,10 +171,64 @@ def _parse_input(obj1, obj2, operator):
...
+    else:
+        opSel = None
+    log_dct = {"operator": operator,
+               "base": baseObj.__class__.__name__,
+               "base selection": baseObj._selection,
+               "operand": operand.__class__.__name__,
+               "operand selection": opSel}
+
+    # Create output object
+    out = baseObj.__class__(dimord=baseObj.dimord)
```
Comment (tensionhead) [inline syncopy/datatype/methods/arithmetic.py:199]: Very neat! What happens in cases like `data1 = data1 + data2` with the underlying hdf5 container(s) of `data1`?

File: syncopy/datatype/methods/arithmetic.py [hunk 3]
```diff
@@ -165,10 +171,64 @@ def _parse_input(obj1, obj2, operator):
...
+            parallel = True
+        except ValueError:
+            parallel = False
+
+    # Perform actual computation
+    opMethod = SpyArithmetic(operand_dat, operand_idxs, operation=operation,
+                             opres_type=opres_type)
+    opMethod.initialize(baseObj,
+                        out._stackingDim,
+                        chan_per_worker=None,
```
Comment (tensionhead) [inline syncopy/datatype/methods/arithmetic.py:231]: Arithmetic should be almost trivially parallelisable over channels no? However probably not really needed..

**Review by tensionhead — COMMENTED** [review]
Looks all really good to me!

File: syncopy/datatype/base_data.py (hunk 1 above)
Comment (pantaray) [reply syncopy/datatype/base_data.py:749]: :D Thanks. Necessity and invention and all that

File: syncopy/datatype/methods/arithmetic.py (hunk 3 above)
Comment (pantaray) [reply syncopy/datatype/methods/arithmetic.py:231]: Yupp, should be. The only thing that's sabotaging it is the possible existence of in-place selections in one (or both) objects. Then the channel parallelization setup becomes complicated. Definitely not impossible, but unless users request it, I thought we could table that one for later

File: syncopy/datatype/methods/arithmetic.py (hunk 2 above)
Comment (pantaray) [reply syncopy/datatype/methods/arithmetic.py:199]: Good catch!!! Works out of the box sequentially, but breaks in parallel.

File: syncopy/datatype/methods/arithmetic.py (hunk 3 above)
Comment (tensionhead) [reply syncopy/datatype/methods/arithmetic.py:231]: Agreed :+1:

File: syncopy/datatype/methods/arithmetic.py (hunk 2 above)
Comment (pantaray) [reply syncopy/datatype/methods/arithmetic.py:199]: Correction - works in parallel, my tests were just wrong. Sorry for the confusion!

**Pull request merged**